hummingbot/core/data_type/order_book_tracker.py

```diff
@@ -136,11 +139,34 @@ def stop(self):
         if self._order_book_snapshot_router_task is not None:
             self._order_book_snapshot_router_task.cancel()
             self._order_book_snapshot_router_task = None
+        if self._update_last_trade_prices_task is not None:
+            self._update_last_trade_prices_task.cancel()
+            self._update_last_trade_prices_task = None
         if len(self._tracking_tasks) > 0:
             for _, task in self._tracking_tasks.items():
                 task.cancel()
             self._tracking_tasks.clear()
 
+    async def _update_last_trade_prices_loop(self):
+        '''
+        Updates last trade price for all order books, this is fall-back mechanism for when the web socket
+        has not yet started receiving messages or fails for some reasons.
+        '''
+        await wait_til(lambda: len(self.data_source._trading_pairs) == len(self._order_books.keys()))
+        while True:
+            try:
+                for trading_pair, order_book in self._order_books.items():
+                    if order_book.last_trade_price_updated >= time.perf_counter() - (60. * 3):
```

Comment:
1. I believe you want this to only run if `last_trade_price` isn't updated within 3 minutes. I don't see anywhere else `last_trade_price_updated` is updated aside in `c_apply_trade()`(which is only used for simulating trades). I believe `last_trade_price_updated` should be updated later in this block. It should also be updated after `last_trade_price` is also updated over websocket.

2. Also, I can't see where `last_trade_price ` is updated over websocket. I guess you plan to do that in subsequent commit for this PR.

Reply:
`c_apply_trade()` is used for when the bot is listening to trades updates through web socket channel, if the web socket is configured and connected properly, there is no additional work required for this. So the `last_trade_price` is simply whatever the last trade update. I will rename `last_trade_price_updated` to `last_applied_trade` to avoid confusion. `_update_last_trade_prices_loop` is a fall back mechanism for when web_socket is failing, it checks every 3 seconds if `last_trade_price_updated` is over 3 minutes it will use REST API to update the last trade price. The reason `last_trade_price_updated` is not updated here is because it will update the price every 3 seconds til the web socket mechanism kicks in. Also note that REST API is a bit slower than ws API, to avoid concurrency problem, it is designed to be used only as a fall back mechanism.

Reply:
I see, nice job!